### PR TITLE
develop to master

### PR DESCRIPTION
**Pull request by budgetpreneur**

### CODE REVIEW

File: src/Api/PubnubApi/Enum/PushTypeService.cs [hunk 1]
```diff
@@ -3,8 +3,9 @@ namespace PubnubApi
 {
     public enum PNPushType
     {
-        GCM,
```
Comment (davidnub) [inline src/Api/PubnubApi/Enum/PushTypeService.cs]: We need to make sure to document this if we're going to remove it. Will this break current users with this Type? Is it possible to leave it in as backwards compatibility?

File: src/Api/PubnubApi/Pubnub.cs [hunk 2]
```diff
@@ -19,7 +19,7 @@ public class Pubnub
 
         private readonly string instanceId;
 
-        private static string sdkVersion = string.Format("{0}CSharp4.2.0.0", PNPlatform.Get());
+        private static string sdkVersion = string.Format("{0}CSharp4.3.0.0", PNPlatform.Get());
```
Comment (davidnub) [inline src/Api/PubnubApi/Pubnub.cs]: Would be great if we could pull this from somewhere in the future.

File: src/Api/PubnubApi/Enum/PushTypeService.cs (hunk 1 above)
Comment (budgetpreneur) [reply src/Api/PubnubApi/Enum/PushTypeService.cs]: re-added GCM for backward-compatibility. So we will have both FCM and GCM

File: src/Api/PubnubApi/Pubnub.cs (hunk 2 above)
Comment (budgetpreneur) [reply src/Api/PubnubApi/Pubnub.cs]: image
SDK has AssemblyInfo.cs files which needs manual update. AssemblyInfo.cs updates version numbers on the individual Pubnub.dll properties.

**Pull request merged**